dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/metrics/TaskMetrics.java

```diff
@@ -83,11 +84,15 @@ public void incTaskDispatch() {
         taskDispatchCounter.increment();
     }
 
-    public void incTaskInstanceByState(final String state) {
-        if (taskInstanceCounters.get(state) == null) {
+    public void incTaskInstanceByState(final String state, String taskInstanceName) {
+        Counter.Builder builder = taskInstanceCountersBuild.get(state);
+        if (builder == null) {
             return;
         }
-        taskInstanceCounters.get(state).increment();
+        if (MasterConfigStatic.masterConfigStatic.isSupportTaskNameTagMetric()) {
```

Comment:
https://github.com/apache/dolphinscheduler/blob/1ac76e1cbbf25156f3b40dc5035ef7b15dfc9bdf/dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql#L468-L503

`task name` is not identical, aggregating metrics by `task name` may lead to miscounting. `id` is identical for task definition, however, I haven't found a way for users to get `task definition id` easily from web ui.